var/spack/repos/builtin/packages/xorgproto/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Xorgproto(MesonPackage):
+    """X Window System unified protocol definitions replacing standalone protos"""
+
+    homepage = "https://gitlab.freedesktop.org/xorg/proto/xorgproto"
+    url = "https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/archive/xorgproto-2024.1/xorgproto-xorgproto-2024.1.tar.bz2"
+
+    maintainers("teaguesterling")
+
+    license("OTHER", checked_by="teaguesterling")
+
+    version("2024.1", sha256="3959b2d17d86dd9d165dc24d26f372ca64f27127cd381739366ba8383a6cd51a")
+
```

Comment:
There is one option, `legacy`, https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/blob/master/meson_options.txt?ref_type=heads, which avoids installing some rarely used protocols, https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/blob/master/meson.build?ref_type=heads#L88.

Reply:
We should add this as a variant, yeah?

Reply:
I think that would be preferred, but we could also just by default run with `legacy=OFF` (probably explicitly in `meson_args`).

We'd have to see what we would actually lose without the legacy protocols... It doesn't look like we would be restricting anything much.
```
$ for i in 'evieproto' 'fontcacheproto' 'lg3dproto' 'printproto' 'trapproto' 'windowswmproto' 'xcalibrateproto' 'xf86miscproto' 'xf86rushproto' 'xproxymngproto' ; do echo -n "$i: " ; spack dependents -t $i ; done
evieproto: No dependents
fontcacheproto: libxfontcache
lg3dproto: No dependents
printproto: grace  libxp  libxprintapputil  libxprintutil  xphelloworld  xplsprinters  xprehashprinterlist
trapproto: libxtrap  xtrap
windowswmproto: libwindowswm
xcalibrateproto: No dependents
xf86miscproto: libxxf86misc
xf86rushproto: No dependents
xproxymngproto: No dependents
```

Reply:
Let's just drop these legacy protocols here.